### PR TITLE
Bug 1939731: Add PodDisruptionBudget for image-registry

### PR DESCRIPTION
**Pull request by dmage**

### CODE REVIEW

**Comment by wking** [comment]
[e2e job][1] looks good:

```console
$ curl -s https://gcsweb-ci.apps.ci.l2s4.p1.openshiftapps.com/gcs/origin-ci-test/pr-logs/pull/openshift_cluster-image-registry-operator/671/pull-ci-openshift-cluster-image-registry-operator-master-e2e-aws/1372999475592695808/artifacts/e2e-aws/gather-must-gather/artifacts/must-gather.tar | tar -xOz registry-build02-ci-openshift-org-ci-op-wbgvnq2x-stable-sha256-e8636e07a5dd82c49cf917f78d8fd877765a9c899b1b2848ab6b2ad57129cdce/namespaces/openshift-image-registry/policy/poddisruptionbudgets.yaml
---
apiVersion: policy/v1beta1
items:
- apiVersion: policy/v1beta1
  kind: PodDisruptionBudget
  metadata:
    annotations:
      imageregistry.operator.openshift.io/checksum: sha256:fbd92701d7a2b26e7770e68fe98498809b39e955dd4b69d78db4b07762ac55e6
    creationTimestamp: "2021-03-19T20:18:53Z"
    generation: 1
...
```

[1]: https://prow.ci.openshift.org/view/gs/origin-ci-test/pr-logs/pull/openshift_cluster-image-registry-operator/671/pull-ci-openshift-cluster-image-registry-operator-master-e2e-aws/1372999475592695808

File: pkg/resource/poddisruptionbudget.go [hunk 1]
```diff
@@ -0,0 +1,88 @@
...
+		client: client,
+	}
+}
+
+func (gpdb *generatorPodDisruptionBudget) Type() runtime.Object {
+	return &policyv1beta1.PodDisruptionBudget{}
+}
+
+func (gpdb *generatorPodDisruptionBudget) GetNamespace() string {
+	return "openshift-image-registry"
```
Comment (ricardomaraschini) [inline pkg/resource/poddisruptionbudget.go:34]: `defaults.ImageRegistryOperatorNamespace` ?

File: pkg/resource/poddisruptionbudget.go [hunk 2]
```diff
@@ -0,0 +1,88 @@
...
+}
+
+func (gpdb *generatorPodDisruptionBudget) GetName() string {
+	return "image-registry"
```
Comment (ricardomaraschini) [inline pkg/resource/poddisruptionbudget.go:38]: `defaults.ImageRegistryName` ?

**Review by ricardomaraschini — CHANGES_REQUESTED** [review]
Just two small things, otherwise LGTM.

File: pkg/resource/poddisruptionbudget.go (hunk 1 above)
Comment (dmage) [reply pkg/resource/poddisruptionbudget.go]: Updated, but it's unclear if we have benefits from this constant.

File: pkg/resource/poddisruptionbudget.go (hunk 2 above)
Comment (dmage) [reply pkg/resource/poddisruptionbudget.go:38]: No, this constant is documented as for the deployment. If we want to have named constants, we need to create a new one for PDB. But I'd rather use a string literal.

**Review by ricardomaraschini — APPROVED** [review]
/lgtm

**Pull request merged**